cns/restserver/util.go

```diff
@@ -622,6 +622,40 @@ func (service *HTTPRestService) SendNCSnapShotPeriodically(ncSnapshotIntervalInM
 	}
 }
 
+func (service *HTTPRestService) validateIpConfigRequest(ipConfigRequest cns.GetIPConfigRequest) (int, string) {
+	if service.state.OrchestratorType != cns.KubernetesCRD {
+		return UnsupportedOrchestratorType, fmt.Sprintf("ReleaseIPConfig API supported only for kubernetes orchestrator")
+	}
+
+	if ipConfigRequest.OrchestratorContext == nil {
+		return EmptyOrchastretorContext, fmt.Sprintf("OrchastratorContext is not set in the req: %+v", ipConfigRequest)
+	}
+
+	// retrieve podinfo  from orchestrator context
+	var podInfo cns.KubernetesPodInfo
+	if err := json.Unmarshal(ipConfigRequest.OrchestratorContext, &podInfo); err != nil {
+		return UnsupportedOrchastretorContext, err.Error()
+	}
+
+	return Success, ""
+}
+
+func (service *HTTPRestService) populateIpConfigInfoFromNCUntransacted(ncId string, ipConfiguration *cns.IPConfiguration) error {
+	var (
+		ncStatus containerstatus
+		exists   bool
+	)
+
+	if ncStatus, exists = service.state.ContainerStatus[ncId]; !exists {
+		return fmt.Errorf("Failed to get NC Configuration for NcId: %s", ncId)
+	}
+
+	ipConfiguration.DNSServers = ncStatus.CreateNetworkContainerRequest.IPConfiguration.DNSServers
```

Comment:
Can we add a check to the cnsclient_test to verify these changes from CNI perspective?